Scripts/Runtime/Data/IDProviderAtomicBurstable.cs

```diff
@@ -0,0 +1,76 @@
+using Anvil.CSharp.Data;
+using System.Threading;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// An <see cref="IDProvider"/> that is compatible with Burst and will get the next ID atomically so it is
+    /// safe to use in threaded context.
+    /// </summary>
+    [BurstCompatible]
+    public readonly unsafe struct IDProviderAtomicBurstable
+    {
+        public const uint DEFAULT_SUPPLY_WARNING_THRESHOLD = uint.MaxValue - 1_000_000;
+        
+        /// <summary>
+        /// The threshold to signify the ID supply is near exhaustion
+        /// </summary>
+        public readonly uint SupplyWarningThreshold;
+        
+        [NativeDisableUnsafePtrRestriction] private readonly uint* m_IDPointer;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="IDProviderAtomicBurstable"/>.
+        /// </summary>
+        /// <remarks>
+        /// NOTE: Because this is a struct, it could be created by using the default constructor which will fail.
+        /// Debug.Asserts will catch this at runtime when used.
+        /// It is expected to create this struct and pass in an explicit supply warning threshold instead, the
+        /// <see cref="DEFAULT_SUPPLY_WARNING_THRESHOLD"/> is provided to make this easy to do so.
+        ///
+        /// ALSO NOTE: Because this must be compatible with Burst, there is no event to signify when the
+        /// <see cref="SupplyWarningThreshold"/> has been met like in <see cref="IDProvider"/>. Instead, you must
+        /// poll this struct periodically and call <see cref="HasIDExceededSupplyWarningThreshold"/>.
+        /// </remarks>
+        /// <param name="supplyWarningThreshold">The threshold to signify the ID supply is near exhaustion</param>
+        public IDProviderAtomicBurstable(uint supplyWarningThreshold)
```

Comment:
Set a default for the parameter like `IDProvider` does

Reply:
That doesn't work with structs. (in this version of C# anyway). 

If you create the new instance and don't explicitly pass in a value, it will call the default constructor with no parameters. 

The `Debug.Assert` in `GetNextID` traps for this. 

Using the system in #265 could make this nicer by making the constructor internal and having the creation go through the System. But for now, the docs handle it.

Reply:
ah right, ok. Good catch